Description: Do not use SNAPCRAFT_BUILD_INFO to detect a test environment.
 .
 That is simply incorrect and also hits the normal build environment,
 causing a build failure.
 .
Forwarded: https://phabricator.services.mozilla.com/D274753
Author: Nathan Pratta Teodosio <nteodosio@ubuntu.com>

--- a/toolkit/crashreporter/tools/symbolstore.py
+++ b/toolkit/crashreporter/tools/symbolstore.py
@@ -802,14 +802,11 @@
 
             if "asan" not in perfherder_extra_options.lower():
                 print(
                     "PERFHERDER_DATA: %s" % json.dumps(perfherder_data), file=sys.stderr
                 )
-                if (
-                    "MOZ_AUTOMATION" in os.environ
-                    or "SNAPCRAFT_BUILD_INFO" in os.environ
-                ):
+                if "MOZ_AUTOMATION" in os.environ or "MOZILLA_SNAP_TEST" in os.environ:
                     upload_dir = Path(os.environ.get("UPLOAD_DIR"))
                     upload_dir.mkdir(parents=True, exist_ok=True)
                     upload_path = upload_dir / "perfherder-data-compiler-metrics.json"
                     with upload_path.open("w", encoding="utf-8") as f:
                         json.dump(perfherder_data, f)

